### ISSUE TITLE
WhiteSource Unified Agent is taking extraordinarily long time to complete or eventually timeout in maven scan.

### ISSUE THREAD

**Issue by MisterDurden** [whitesource/unified-agent-distribution#35]
Hi,
I am trying to do a scan for Java-Maven based project but its getting timeout all the time.
The issue is discussed here:
https://whitesource.atlassian.net/wiki/spaces/WSK/pages/2481225794/Unsupported+Maven+Dependency+Plugin+versions+3.0.x+3.1.x+3.2.x+Workaround

But both the workarounds are not working for us. I am still facing the same issue and not able to complete the scan.
Is there any fix being released in the upcoming days?
The scan is working fine on my local but getting failed on the Jenkins machine.

**Comment by LenaKleyner** [comment]
Hi @MisterDurden, we are looking into this and hope to introduce a better solution soon.

**Comment by MisterDurden** [comment]
We tried using the maven.additionalArguments=-Dverbose=false flag in the config file but still it wasn't able to override the verbose flag in maven command during the scan.
Is there any update on this?

**Comment by avinoamw1** [comment]
Hi @MisterDurden ,
I suggest retrying next week using version 22.4.2 of the Unified Agent, that should be out on May 15th.

**Comment by MisterDurden** [comment]
It still isn't working.
We have our own custom settings.xml with some proxy settings which is running fine for other maven goals apart from the Whitesource scan one.


```
- Trying to resolve MAVEN dependencies
- topFolder = /somelocation/
- Command - executeProcess - error in execute command 'mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:tree -DoutputFile=whitesource_mvn_dependency_tree.txt -Dverbose -DoutputType=text -T1 -s=someLocation/settings.xml -B', Exit Status 1
 - Command - executeProcess - error in execute command 'mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:tree -DoutputFile=whitesource_mvn_dependency_tree.txt -Dverbose -DoutputType=text -T1 -s=someLocation/settings.xml', Exit Status 1
- maven.resolveDependencies = true, but mvn dependency:tree failed
- Command - executeProcess - Timeout while running command mvn dependency:tree -DoutputFile=whitesource_mvn_dependency_tree.txt -Dverbose -DoutputType=text -T1 -s=someLocation/settings.xml -B, timeout = 900 seconds
 - Command - executeProcess - error executing command 'mvn dependency:tree -DoutputFile=whitesource_mvn_dependency_tree.txt -Dverbose -DoutputType=text -T1 -s=someLocation/settings.xml -B' destroying process
 - Command - executeProcess - Timeout while running command mvn dependency:tree -DoutputFile=whitesource_mvn_dependency_tree.txt -Dverbose -DoutputType=text -T1 -s=someLocation/settings.xml, timeout = 900 seconds
```


The pipeline is running inside a docker container for Ubuntu and the whitesource maven config that I am using is: 


```
maven.additionalArguments=-s=someLocation/settings.xml
maven.resolveDependencies=true
maven.aggregateModules=false
```

I have tried using maven.runPreStep=false but it still it didn't work.
Our maven repository access is proxied through our internal repo, so using the custom settings.xml is important.
Can you tell if there is any other place where we can specify proxy settings in unified-agent?

**Comment by LenaKleyner** [comment]
@MisterDurden this was indeed resolved as part of release 22.4.2 but if your issue still persists could you please open a ticket in our [support portal](https://whitesourcesoftware.force.com/SupportPortal/s/login/) with all the relevant information (the Unified Agent execution command, its settings, and the output log) so we could further investigate this?
